caas/kubernetes/cloud/credential.go

```diff
@@ -345,3 +345,18 @@ func SupportedNonLegacyAuthTypes() cloud.AuthTypes {
 	sort.Sort(ats)
 	return ats
 }
+
+// PatchCloudCredentialForCloudSpec patches the credential with the provided k8s secret token.
+func PatchCloudCredentialForCloudSpec(cred cloud.Credential, token string) (cloud.Credential, error) {
```

Comment:
I think the name can be better, maybe `AddTokenToCredential` or `UpdateCredentialWithToken`